src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts

```diff
@@ -185,46 +213,89 @@ export const load = async ({
       spaceID: CONTENTFUL_SPACE_ID,
       token: CONTENTFUL_DELIVERY_API_TOKEN,
     });
-    const data = await client.fetch<ServiceGroupCollectionQuery>(printQuery(query), {
+
+    const baseData = await client.fetch<ServiceGroupQuery>(printQuery(baseQuery), {
       variables: { metadataID },
     });
-    if (!data) break fetchData;
-    const [serviceGroup] = data?.serviceGroupCollection?.items ?? [];
+    if (!baseData) break fetchData;
+    const [serviceGroup] = baseData?.serviceGroupCollection?.items ?? [];
     if (!serviceGroup) break fetchData;
```

Comment:
since serviceGroup is always coalescing to `[]` when `undefined` or `null`, won't this this `break` statement never occur?

Reply:
The destructuring assignment here assigns `serviceGroup` to the first element of `baseData?.serviceGroupCollection?.items ?? []`. If `baseData?.serviceGroupCollection?.items` is `null` or `undefined`, `serviceGroup` will also be `undefined` because it's assigned to the first element of an empty array.

Reply:
got it. missed the destructuring bit